### PR TITLE
chore: Add support for "interim" hint in unpublished DIDs

### LINKED ISSUE #582: Add support for "interim" hint in unpublished DIDs
If domain is specified for document handler create an equivalent ID with "interim:domain" hint for the result of create operation (unpublished DID document)

### PR DESCRIPTION
**Pull request by sandrask**
If domain is specified for document handler create an equivalent ID with "interim:domain" hint for the result of create operation (unpublished DID document)

Closes #582

Signed-off-by: Sandra Vrtikapa <sandra.vrtikapa@securekey.com>

### CODE REVIEW

**Review by fqutishat — APPROVED** [review]

**Review by bstasyszyn — APPROVED** [review]

File: pkg/dochandler/handler.go [hunk 1]
```diff
@@ -139,7 +157,24 @@ func (r *DocumentHandler) getCreateResponse(op *operation.Operation, pv protocol
...
+	ti[document.PublishedProperty] = false
+
+	id := r.namespace + docutil.NamespaceDelimiter + op.UniqueSuffix
+
+	// if optional domain is specified we should set equivalent id with domain hint for interim/unpublished documents
+	if r.domain != "" {
+		id = r.namespace + docutil.NamespaceDelimiter + "interim" + docutil.NamespaceDelimiter + op.UniqueSuffix
+
+		equivalentID := r.namespace + docutil.NamespaceDelimiter +
+			"interim" + docutil.NamespaceDelimiter + r.domain + docutil.NamespaceDelimiter + op.UniqueSuffix
```
Comment (troyronda) [inline pkg/dochandler/handler.go:170]: I think the label portion (e.g., "interim:") is DID method specific.

Comment (troyronda) [reply pkg/dochandler/handler.go:170]: We should make this configurable/settable by the concrete implementation downstream project.

Comment (sandrask) [reply pkg/dochandler/handler.go:170]: It will only show up if domain is specified. Domain is not specified in other cases.

Comment (sandrask) [reply pkg/dochandler/handler.go:170]: I will make it configurable (same as domain).

**Pull request merged**